scvi/module/_jaxpeakvae.py

```diff
@@ -0,0 +1,201 @@
+from typing import Dict
+
+import jax
+import jax.numpy as jnp
+import numpyro.distributions as dist
+from flax import linen as nn
+
+from scvi import REGISTRY_KEYS
+from scvi.module.base import JaxBaseModuleClass, LossRecorder
+
+from ._jaxvae import Dense
+
+
+class FlaxEncoder(nn.Module):
+    n_input: int
+    n_latent: int
+    n_hidden: int
+    dropout_rate: int
+
+    def setup(self):
+        self.dense1 = Dense(self.n_hidden)
+        self.dense2 = Dense(self.n_hidden)
+        self.dense3 = Dense(self.n_latent)
+        self.dense4 = Dense(self.n_latent)
+
+        self.layernorm1 = nn.LayerNorm()
+        self.layernorm2 = nn.LayerNorm()
+        self.dropout1 = nn.Dropout(self.dropout_rate)
+        self.dropout2 = nn.Dropout(self.dropout_rate)
+
+    def __call__(self, x: jnp.ndarray, training: bool = False):
+        is_eval = not training
+
+        x_ = jnp.log1p(x)
+
+        h = self.dense1(x_)
+        h = self.layernorm1(h)
+        h = nn.leaky_relu(h)
+        h = self.dropout1(h, deterministic=is_eval)
+        h = self.dense2(h)
+        h = self.layernorm2(h)
+        h = nn.leaky_relu(h)
+        h = self.dropout2(h, deterministic=is_eval)
+
+        mean = self.dense3(h)
+        log_var = self.dense4(h)
+
+        return mean, jnp.exp(log_var)
+
+
+class FlaxDecoder(nn.Module):
+    n_input: int
+    dropout_rate: float
+    n_hidden: int
+    n_output: int = None
+
+    def setup(self):
+        self.dense1 = Dense(self.n_hidden)
+        self.dense2 = Dense(self.n_hidden)
+        self.dense3 = Dense(self.n_hidden)
+        self.dense4 = Dense(self.n_hidden)
+        if self.n_output is None:
```

Comment:
do we need n_input if it's inferred? Seems like we just need n_output to be defined for every instance